sway-core/src/language/ty/declaration/declaration.rs

```diff
@@ -466,8 +484,32 @@ impl TyDeclaration {
         }
     }
 
-    /// friendly name string used for error reporting.
-    pub fn friendly_name(&self) -> &'static str {
+    /// friendly type name string used for error reporting.
```

Comment:
The doc comments should probably explain the difference between `friendly_name and `friendly_type_name`.